src/storage/mutate/AddEdgesProcessor.cpp

```diff
@@ -386,13 +372,24 @@ std::vector<std::string> AddEdgesProcessor::indexKeys(
  * ifNotExist_ is true. Only keep the first one when edgeKey is same
  * ifNotExist_ is false. Only keep the last one when edgeKey is same
  */
-void AddEdgesProcessor::deleteDupEdge(std::vector<cpp2::NewEdge>& edges) {
+nebula::cpp2::ErrorCode AddEdgesProcessor::deleteDupEdge(std::vector<cpp2::NewEdge>& edges) {
+  auto code = nebula::cpp2::ErrorCode::SUCCEEDED;
   std::unordered_set<std::string> visited;
   visited.reserve(edges.size());
   if (ifNotExists_) {
     auto iter = edges.begin();
     while (iter != edges.end()) {
       auto edgeKeyRef = iter->key_ref();
+      if (!NebulaKeyUtils::isValidVidLen(
+              spaceVidLen_, edgeKeyRef->src_ref()->getStr(), edgeKeyRef->dst_ref()->getStr())) {
+        LOG(ERROR) << "Space " << spaceId_ << " vertex length invalid, "
+                   << "space vid len: " << spaceVidLen_
+                   << ", edge srcVid: " << *edgeKeyRef->src_ref()
+                   << ", dstVid: " << *edgeKeyRef->dst_ref() << ", ifNotExists_: " << std::boolalpha
+                   << ifNotExists_;
+        code = nebula::cpp2::ErrorCode::E_INVALID_VID;
```

Comment:
May be not need a variable code in this function, just return `nebula::cpp2::ErrorCode::E_INVALID_VID;` will be ok

Reply:
good idea